### PR TITLE
[libtasn1] Add Windows support, update to 4.17

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**
Enables windows build

- Which triplets are supported/not supported? Have you updated the CI baseline?

static-md build is unsupported. CI baseline updated.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by wrobelda** [comment]
Currently stuck at:
```
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_parser2tree referenced in function _main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_print_structure referenced in function _main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_create_element referenced in function _main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_delete_structure referenced in function _main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_write_value referenced in function _main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_der_coding referenced in function _main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp__asn1_strerror referenced in function _main
.libs\asn1Coding.exe : fatal error LNK1120: 7 unresolved externals
asn1Parser.obj : error LNK2019: unresolved external symbol __imp__asn1_parser2tree referenced in function _main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp__asn1_parser2array referenced in function _main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp__asn1_delete_structure referenced in function _main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp__asn1_strerror referenced in function _main
.libs\asn1Parser.exe : fatal error LNK1120: 4 unresolved externals
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fno-common'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-show-option'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
cl : Command line warning D9002 : ignoring unknown option '-fdiagnostics-color=always'
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_parser2tree referenced in function _main
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_print_structure referenced in function _simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_create_element referenced in function _simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_delete_structure referenced in function _main
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_der_decoding2 referenced in function _simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_der_decoding referenced in function _simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp__asn1_strerror referenced in function _main
.libs\asn1Decoding.exe : fatal error LNK1120: 7 unresolved externals
make[2]: Leaving directory '/cygdrive/c/vcpkg/buildtrees/libtasn1/x86-windows-dbg/src'
make[1]: Leaving directory '/cygdrive/c/vcpkg/buildtrees/libtasn1/x86-windows-dbg'
```

**Comment by wrobelda** [comment]
OK, so what I see is:
```
/bin/sh ../libtool  --tag=CC   --mode=link compile cl.exe -fno-common -Wall -fdiagnostics-show-option -fdiagnostics-color=always   -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1  -LC:/vcpkg/installed/x86-windows/debug/lib/manual-link/ -lgettimeofday -lgetopt -o asn1Decoding.exe asn1Decoding.obj benchmark.obj ../lib/libtasn1.la -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32
```
followed by:
```
libtool: link: compile cl.exe -fno-common -Wall -fdiagnostics-show-option -fdiagnostics-color=always -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -RTC1 -o .libs/asn1Coding.exe asn1Coding.obj  -LC:/vcpkg/installed/x86-windows/debug/lib/manual-link/ ../lib/.libs/tasn1.lib -lgettimeofday -lgetopt -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32
```
The missing symboles are in `../lib/libtasn1.la`, which is what lib tool asks for, yet the the actual command issued asks for `../lib/.libs/tasn1.lib`, so something is messing it in the process.

**Comment by wrobelda** [comment]
Most interestingly, the `../lib/.libs/tasn1.lib` ends up in the middle of my additional, explicit `LDFLAGS`:
`LC:/vcpkg/installed/x86-windows/debug/lib/manual-link/ ../lib/.libs/tasn1.lib -lgettimeofday`

**Comment by wrobelda** [comment]
I tried setting the LDFLAGS using env variable instead (`set(ENV{LDFLAGS} "$ENV{LDFLAGS} -L${CURRENT_INSTALLED_DIR}/lib/manual-link -lgettimeofday -lgetopt"`), but it is being ignored.

**Comment by wrobelda** [comment]
OK, so I tried setting the LDFLAGS using CONFIGURE_ENVIRONMENT_VARIABLES, but it doesn't make a difference: the `../lib/libtasn1.la `somehow ends up as` ../lib/.libs/tasn1.lib` , in the middle of my `LDFLAGS`

This is clearly some bug.

**Comment by Neumann-A** [comment]
That is probably a bug in a `Makefile` of the project.

**Comment by wrobelda** [comment]
OK, so this is till to do with overloading LDFLAGS. From the build log:

```
*** Warning: linker path does not have real file for library -lgetopt.
*** I have the capability to make that library automatically link in when
*** you link to this library.  But I can only do this if you have a
*** shared version of the library, which you do not appear to have
*** because I did check the linker path looking for a file starting
*** with getopt but no candidates were found. (...for file magic test)
*** The inter-library dependencies that have been dropped here will be
*** automatically added whenever a program is linked with this library
*** or is declared to -dlopen it.

*** Since this library must not contain undefined symbols,
*** because either the platform does not support them or
*** it was explicitly requested with -no-undefined,
*** libtool will only create a static version of it.
```

Setting LDFLAGS with `set(LDFLAGS "${LDFLAGS} -lgetopt -L${CURRENT_INSTALLED_DIR}/lib/manual-link -lgettimeofday")` ends up with following lines in config.log:

```
LDFLAGS=' -lgetopt -LC:/vcpkg/installed/x86-windows/lib/manual-link -lgettimeofday'
```

Not setting it will get:

```
LDFLAGS='-LC:/vcpkg/installed/x86-windows/debug/lib -LC:/vcpkg/installed/x86-windows/debug/lib/manual-link'
```

So:
a) LDFLAGS already comes with manual-link preconfigured (this could be documented somewhere)
b) appending to LDFLAGS seems impossible, as back-referencing ${LDFLAGS} or ${VCPKG_LINKER_FLAGS} doesn't work

**Comment by wrobelda** [comment]
Given how vcpkg already configured LDFLAGS as required, I only needed to set the LIBS. 

This was also problematic, given how setting the environment variable would override vcpkg's own list of libraries (`LIBS='-lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32'`).

I worked around by passing LIBS as a regular option to vcpkg_configure_make and have it use shell to expand the environment LIB variable: `set(LIBS "$LIBS -lgettimeofday -lgetopt")`

**Comment by wrobelda** [comment]
Build still fails on CI/CD, but I cannot reproduce those errors locally on my Windows dev box:

```
libtool: compile:  /cygdrive/d/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c -o strverscmp.obj
strverscmp.c
D:\buildtrees\libtasn1\x86-windows-dbg\lib\gl\stdint.h(77): fatal error C1083: Cannot open include file: '': No such file or directory
```

**Comment by NancyLi1013** [comment]
The latest failures on Windows:

From build-x64-windows-dbg-err.log
```
ar-lib: unknown action specified
make[5]: *** [Makefile:687: libgnu.la] Error 1
make[4]: *** [Makefile:738: all-recursive] Error 1
make[3]: *** [Makefile:638: all] Error 2
make[2]: *** [Makefile:808: all-recursive] Error 1
make[1]: *** [Makefile:655: all-recursive] Error 1
make: *** [Makefile:584: all] Error 2
```
Could you please look into this? @wrobelda

**Comment by wrobelda** [comment]
OK, I am stuck, tbh.

The build fails differently for x86 and x64 platforms. On x86 I am seeing:

```
make[5]: Entering directory '/c/vcpkg/buildtrees/libtasn1/x86-windows-dbg/lib/gl'
Makefile:721: update target 'c-ctype.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c' object='c-ctype.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o c-ctype.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
Makefile:721: update target 'strverscmp.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c' object='strverscmp.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o strverscmp.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c  -DDLL_EXPORT -DPIC -o .libs/c-ctype.obj
c-ctype.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c  -DDLL_EXPORT -DPIC -o .libs/strverscmp.obj
strverscmp.c
Makefile:687: update target 'libgnu.la' due to: c-ctype.lo strverscmp.lo
/bin/sh ../../libtool  --tag=CC   --mode=link /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1  -no-undefined -LC:/vcpkg/installed/x86-windows/debug/lib -LC:/vcpkg/installed/x86-windows/debug/lib/manual-link -o libgnu.la  c-ctype.lo strverscmp.lo -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32 -lgettimeofday -lgetopt
libtool: link: lib.exe cru .libs/gnu.lib .libs/c-ctype.obj .libs/strverscmp.obj 
Microsoft (R) Library Manager Version 14.28.29337.0
Copyright (C) Microsoft Corporation.  All rights reserved.

LINK : fatal error LNK1181: cannot open input file 'cru'
make[5]: Leaving directory '/c/vcpkg/buildtrees/libtasn1/x86-windows-dbg/lib/gl'
```

And on x64:

```
make[5]: Entering directory '/c/vcpkg/buildtrees/libtasn1/x64-windows-dbg/lib/gl'
Makefile:721: update target 'c-ctype.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c' object='c-ctype.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o c-ctype.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
Makefile:721: update target 'strverscmp.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c' object='strverscmp.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o strverscmp.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c -o strverscmp.obj
strverscmp.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c -o c-ctype.obj
c-ctype.c
Makefile:687: update target 'libgnu.la' due to: c-ctype.lo strverscmp.lo
/bin/sh ../../libtool  --tag=CC   --mode=link /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1  -no-undefined -LC:/vcpkg/installed/x64-windows/debug/lib -LC:/vcpkg/installed/x64-windows/debug/lib/manual-link -o libgnu.la  c-ctype.lo strverscmp.lo -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32 -lgettimeofday -lgetopt
libtool: link: /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib lib.exe cru .libs/libgnu.a  c-ctype.obj strverscmp.obj
make[5]: Leaving directory '/c/vcpkg/buildtrees/libtasn1/x64-windows-dbg/lib/gl'
```
With the error being:
`ar-lib: unknown action specified`

The x64 libtool call seems to reference `ar-lib` twice – could that explain  the "unknown action specified"?
`libtool: link: /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib lib.exe cru .libs/libgnu.a  c-ctype.obj strverscmp.obj`

**Comment by wrobelda** [comment]
@NancyLi1013 all issues are fixed now.

File: ports/libtasn1/portfile.cmake [hunk 1]
```diff
@@ -1,32 +1,47 @@
...
+        ${EXTRA_OPTS}
 )
 
 vcpkg_install_make()
 vcpkg_fixup_pkgconfig()
-vcpkg_copy_pdbs()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/tools ${CURRENT_PACKAGES_DIR}/debug/share)
```
Comment (NancyLi1013) [inline ports/libtasn1/portfile.cmake:46]: Why do you remove tools here?

Comment (wrobelda) [reply ports/libtasn1/portfile.cmake:46]: These are basic ASN.1 DER decoder/coder/parser tools that are not related to the library itself but rather more of an example of how to use it.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now. thanks for your work @wrobelda.

File: ports/libtasn1/portfile.cmake [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (dg0yt) [inline ports/libtasn1/portfile.cmake:47]: Not sure if vcpkg will ever support spaces in file paths but I would generally recommend double quotes for file path expessions in cmake. (Multiple spot in this file.)

Comment (wrobelda) [reply ports/libtasn1/portfile.cmake:47]: Frankly speaking, I stopped doing this after noticing that absolute vast majority of portfiles don't care. It was never requested from me, either, upon code reviewing. @NancyLi1013 do you have any official policy about it?

Comment (NancyLi1013) [reply ports/libtasn1/portfile.cmake:47]: @dg0yt thanks for your attention to spaces in file paths. Your suggestion is also reasonable. We can suggest contributors to do this, but it is not required. As you can see, there are still some ports without handling the spaces. So it is optional.
@wrobelda Currently, we have no official policy or document about this.

**Comment by strega-nil-ms** [comment]
LGTM! Thanks @wrobelda :)

**Comment by Neumann-A** [comment]
CI baseline regression in static builds:
```
   Creating library asn1Parser.lib and object asn1Parser.exp
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2tree referenced in function main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2array referenced in function main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_delete_structure referenced in function main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_strerror referenced in function main
asn1Parser.exe : fatal error LNK1120: 4 unresolved externals
   Creating library asn1Decoding.lib and object asn1Decoding.exp
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2tree referenced in function main
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_print_structure referenced in function simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_create_element referenced in function simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_delete_structure referenced in function main
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_der_decoding2 referenced in function simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_der_decoding referenced in function simple_decode
asn1Decoding.obj : error LNK2019: unresolved external symbol __imp_asn1_strerror referenced in function main
asn1Decoding.exe : fatal error LNK1120: 7 unresolved externals
```

**Comment by wrobelda** [comment]
> CI baseline regression in static builds:

Just realized that the most recent CI/CD build was `libtasn1:x64-windows-static:    cascade:`. I'll look into it.

**Comment by wrobelda** [comment]
> CI baseline regression in static builds:

@Neumann-A OK, I thought it was about the `x64-windows-static`, which refuses to build due to `getopt` issue, which in turn explains the `cascade` in CI/CD .

Your log section corresponds to `x64-windows-static-md`, which I already added to `scripts/ci.baseline.txt` as failing.  I did not have any more time to work on it, and with `x64-windows-static-md` being an unofficial triplet, I figured it was worth pushing the changes as is.

**Comment by JackBoosY** [comment]
```
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2tree referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_print_structure referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_create_element referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_delete_structure referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_write_value referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_der_coding referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_strerror referenced in function main
```
Please notice the prefix `__imp_`, it must use the dynamic declearation of the functions.

**Pull request merged**